### PR TITLE
refactor: remove unused param from legacy pubkey interface

### PR DESCRIPTION
**Pull request by Bushstar**
Unused param present in legacy pubkey manager interface. This param will not be used and should be removed to prevent unintended usage.

### CODE REVIEW

File: src/wallet/scriptpubkeyman.cpp [hunk 1]
```diff
@@ -1385,9 +1385,10 @@ void LegacyScriptPubKeyMan::ReturnDestination(int64_t nIndex, bool fInternal, co
     WalletLogPrintf("keypool return %d\n", nIndex);
 }
 
-bool LegacyScriptPubKeyMan::GetKeyFromPool(CPubKey& result, const OutputType type, bool internal)
+bool LegacyScriptPubKeyMan::GetKeyFromPool(CPubKey& result, const OutputType type)
 {
     assert(type != OutputType::BECH32M);
+    const bool internal{};
     if (!CanGetAddresses(internal)) {
```
Comment (Sjors) [inline src/wallet/scriptpubkeyman.cpp:1392]: The following style is more consistent with what we do in other places:
'''cpp
if (!CanGetAddresses(/*internal*/=false)) {
``

**Comment by Sjors** [comment]
Concept ACK

File: src/wallet/scriptpubkeyman.cpp (hunk 1 above)
Comment (Bushstar) [reply src/wallet/scriptpubkeyman.cpp:1392]: There's three uses of internal so defining a local bool seemed cleaner. Update all three locations in the style you linked or leave it as it is?

Comment (Bushstar) [reply src/wallet/scriptpubkeyman.cpp]: Updated.

File: src/wallet/scriptpubkeyman.cpp [hunk 2]
```diff
@@ -1385,21 +1385,21 @@ void LegacyScriptPubKeyMan::ReturnDestination(int64_t nIndex, bool fInternal, co
     WalletLogPrintf("keypool return %d\n", nIndex);
 }
 
-bool LegacyScriptPubKeyMan::GetKeyFromPool(CPubKey& result, const OutputType type, bool internal)
+bool LegacyScriptPubKeyMan::GetKeyFromPool(CPubKey& result, const OutputType type)
 {
     assert(type != OutputType::BECH32M);
-    if (!CanGetAddresses(internal)) {
+    if (!CanGetAddresses(/* internal */ false)) {
```
Comment (fanquake) [inline src/wallet/scriptpubkeyman.cpp:1391]:
```suggestion
    if (!CanGetAddresses(/*internal=*/ false)) {
```

Comment (Bushstar) [reply src/wallet/scriptpubkeyman.cpp]: Updated as per the suggestion.

**Comment by fanquake** [comment]
```bash
        if (!ReserveKeyFromKeyPool(nIndex, keypool, /*internal=*/ false) && !m_storage.IsWalletFlagSet(WALLET_FLAG_DISABLE_PRIVATE_KEYS)) {
                                                    ^
./wallet/scriptpubkeyman.h:353:73: note: 'fRequestedInternal' declared here
    bool ReserveKeyFromKeyPool(int64_t& nIndex, CKeyPool& keypool, bool fRequestedInternal);
                                                                        ^
clang-tidy-14 --use-color -p=/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu -quiet /tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/src/wallet/scriptpubkeyman.cpp
wallet/scriptpubkeyman.cpp:1399:53: error: argument name 'internal' in comment does not match parameter name 'fRequestedInternal' [bugprone-argument-comment,-warnings-as-errors]
```

File: src/wallet/scriptpubkeyman.cpp [hunk 3]
```diff
@@ -1385,21 +1385,21 @@ void LegacyScriptPubKeyMan::ReturnDestination(int64_t nIndex, bool fInternal, co
...
+    if (!CanGetAddresses(/*internal=*/ false)) {
         return false;
     }
 
     CKeyPool keypool;
     {
         LOCK(cs_KeyStore);
         int64_t nIndex;
-        if (!ReserveKeyFromKeyPool(nIndex, keypool, internal) && !m_storage.IsWalletFlagSet(WALLET_FLAG_DISABLE_PRIVATE_KEYS)) {
+        if (!ReserveKeyFromKeyPool(nIndex, keypool, /*fRequestedInternal=*/ false) && !m_storage.IsWalletFlagSet(WALLET_FLAG_DISABLE_PRIVATE_KEYS)) {
```
Comment (Sjors) [inline src/wallet/scriptpubkeyman.cpp:1399]: I'm guessing that just dropping the space will fix the CI issue?

```cpp
/*fRequestedInternal=*/false
```

Comment (Bushstar) [reply src/wallet/scriptpubkeyman.cpp:1399]: CI is passing now after rename `internal` to `fRequestedInternal`. CI checks below appear all green to me.

Comment (Sjors) [reply src/wallet/scriptpubkeyman.cpp:1399]: Oh you're right, I missed the last push.

In the majority of cases we don't use a space after `=*/`, but that's not worth touching the PR for imo.

**Review by Sjors — APPROVED** [review]
ACK 1869310f3cfa4ab26b5090d8a4002eefdc84870e

**Review by furszy — APPROVED** [review]
ACK 1869310f3cfa4ab26b5090d8a4002eefdc84870e

**Pull request merged**